src/_modules/data_factory_storage_account/data.tf

```diff
@@ -0,0 +1,20 @@
+data "azurerm_storage_account" "source" {
+  name                = var.storage_accounts.source.name
+  resource_group_name = var.storage_accounts.source.resource_group_name
+}
+
+data "azurerm_storage_account" "target" {
+  name                = var.storage_accounts.target.name
+  resource_group_name = var.storage_accounts.target.resource_group_name
+}
+
+data "azurerm_storage_containers" "this" {
+  count              = var.what_to_migrate.blob.enabled && length(var.what_to_migrate.blob.containers) == 0 ? 1 : 0
+  storage_account_id = data.azurerm_storage_account.source.id
+}
+
+data "azapi_resource_list" "tables" {
```

Comment:
Why this provider? Azurerm includes table storage

Reply:
I couldn't find a better way to list the tables in a storage account with the azurerm provider.
This way, it is not needed to specify each table name to migrate. The module will migrate all of them straight away